pkg/controller/seed-controller-manager/pvwatcher/pvwatcher_controller.go

```diff
@@ -0,0 +1,174 @@
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	ControllerName = "kubermatic_volume_watcher_controller"
+)
+
+type Reconciler struct {
+	log        *zap.SugaredLogger
+	workerName string
+	ctrlruntimeclient.Client
+	recorder record.EventRecorder
+}
+
+// add the controller
+func Add(
+
+	log *zap.SugaredLogger,
+	mgr manager.Manager,
+	numWorkers int,
+	workerName string,
+) error {
+	log = log.Named(ControllerName)
+	reconciler := &Reconciler{
+		log:        log,
+		workerName: workerName,
+		Client:     mgr.GetClient(),
+		recorder:   mgr.GetEventRecorderFor(ControllerName),
+	}
+
+	c, err := controller.New(ControllerName, mgr, controller.Options{
+		Reconciler:              reconciler,
+		MaxConcurrentReconciles: numWorkers,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create controller: %v", err)
+	}
+	// reconcile PVCs in ClaimLost phase only
+	LostClaimPredicates := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			new := e.ObjectNew.(*corev1.PersistentVolumeClaim)
+			return new.Status.Phase == corev1.ClaimLost
+		},
+	}
+
+	if err := c.Watch(&source.Kind{Type: &corev1.PersistentVolumeClaim{}}, &handler.EnqueueRequestForObject{},
+		LostClaimPredicates,
+		predicateutils.ByLabel(resources.AppLabelKey, resources.EtcdStatefulSetName)); err != nil {
+		return fmt.Errorf("failed to create watch fordocker-build PersistentVolumeClaidocker-buildms: %v", err)
+	}
+	return nil
+}
+
+func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	log := r.log.With("request", request)
+	log.Debug("Processing")
+
+	cluster := &kubermaticv1.Cluster{}
+	clusterName := strings.ReplaceAll(request.Namespace, "cluster-", "")
+	if err := r.Get(ctx, types.NamespacedName{Name: clusterName}, cluster); err != nil {
+		if kerrors.IsNotFound(err) {
+			log.Debug("Skipping because the cluster is already gone")
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, err
+	}
+	if cluster.Labels[kubermaticv1.WorkerNameLabelKey] != r.workerName || cluster.Spec.Pause {
+		return reconcile.Result{}, nil
+	}
+	result, err := r.reconcile(ctx, log, request)
+	if err != nil {
+		log.Errorw("Reconciling failed", zap.Error(err))
+		r.recorder.Event(cluster, corev1.EventTypeWarning, "ReconcilingError", err.Error())
+	}
+
+	return result, err
+}
+
+func (r *Reconciler) reconcile(ctx context.Context, log *zap.SugaredLogger, request reconcile.Request) (reconcile.Result, error) {
+	pvc := &corev1.PersistentVolumeClaim{}
+	if err := r.Get(ctx, types.NamespacedName{Name: request.Name, Namespace: request.Namespace}, pvc); err != nil {
+		if kerrors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		log.Errorf("failed to get pvc: %v", err)
```

Comment:
Why is this logged explicitly, when the return value of `reconcile()` will be logged in `Reconcile()` anyway?